src/wp-includes/media.php

```diff
@@ -5564,6 +5676,135 @@ function wp_get_loading_attr_default( $context ) {
 	return 'lazy';
 }
 
+/**
+ * Get loading optimization attributes.
+ *
+ * @since 6.3.0
+ *
+ * @param string $tag_name The tag name.
+ * @param array[] $attr Array of the attributes for the tag.
+ * @param string $context Context for the element for which the loading optimization attribute is requested.
+ *
+ * @return array[] loading optimization attributes.
+ */
+function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
+	global $wp_query;
+
+	$loading_attrs = array();
+
+	// All three parameters are required.
+	if ( empty( $tag_name ) || empty( $attr ) || empty( $context ) ) {
+		return $loading_attrs;
+	}
+
+	// Image tag must have a minimum of width and height.
+	if ( 'img' === $tag_name && ( ! isset( $attr['width'] ) || ! isset( $attr['height'] ) ) ) {
```

Comment:
I think it would be better to remove the `'img' === $tag_name` check from here, since the width and height being provided are general requirements.

If you prefer to keep a check for the tag name, I would suggest instead we return early if `$tag_name` is anything other than `img`, to clarify that for now the function only covers images.
```suggestion
	// For now this function only supports images.
	if ( 'img' === $tag_name ) {
		return $loading_attrs;
	}

	// For any resources, width and height must be provided, to avoid layout shifts.
	if ( empty( $attr['width'] ) || empty( $attr['height'] ) ) {
```

Reply:
Agreed.